testdata/errors.go

```diff
@@ -31,6 +31,12 @@ func g(x int) error {
 		return fmt.Errorf("This %d is too low", x) // MATCH /error strings.*not be capitalized/
 	} else if x == 0 {
 		return fmt.Errorf("XML time") // ok
+	} else if x > 3 {
+		return fmt.Errorf("newlines are fun\n") // MATCH /error strings.*not end with a newline/
+	} else if x > 2 {
+		return fmt.Errorf("Newlines are really fun\n") // MATCH /error strings.*or end with a newline/
+	} else if x > 1 {
+		return fmt.Errorf("newlines are\nespeciallyfun\non a console\n")
 	}
 	return errors.New(`too much stuff.`) // MATCH /error strings.*not end with punctuation/
```

Comment:
Style suggestion. This dangling return covers the case when x == 0, and it was fine when there were 3 cases in total. But with the 3 new cases added here, it becomes quite hard to read.

How about rewriting the entire implementation of `func g(x int) error` to be a single readable switch statement with simple cases over value of `x`.

I know this code is just test code and isn't executed, but it seems harmless to make it more readable (I know it'll be a bigger diff, but I hope that's fine.)